lib/printf/printf-pos.c

```diff
@@ -384,6 +384,7 @@ reswitch:	switch (ch) {
 				goto error;
 			break;
 #endif /* !NO_FLOATING_POINT */
+#if 0
```

Comment:
Why not just remove the code completely?  with a code comment about how %n is totally uncool and will be met with extreme prejudice

Reply:
I mostly wanted to retain the "how to do the function of `%n`" aspect, i.e. recording output positions, in case we want to implement it some other way.

More specifically, while working on the logging stuff, one of the random ideas (pushed off for "later") was that it would be cool to record the start & end of each substituted value and make it accessible as metadata.  For example:

```
"%s: Route %pFX failed to install for table: %u"
=>
"route_notify_owner: Route 10.0.0.0/8 failed to install for table: 1"
=>
[0, 18], [26, 36], [66, 67]
=>
["route_notify_owner", "10.0.0.0/8", "1"]
```

So if you're shoveling your logs into a database you can stuff the substrings into separate fields, which could allow better searching or whatnot.

It's a cool-ish idea on paper, but if you think about it some more it's kinda not that useful so I threw it into the "maybe later, if anyone asks" bin.


Oh and also with `#if 0` the diff to FreeBSD's upstream printf is smaller :smile:

Reply:
Just to be clear, I wouldn't implement the above using `%n`, it just needs similar functionality, so I left the `%n` code there.

Reply:
got it.  I've been going through and removing `#if 0`.  How about changing it to `#ifdef USE_PERCENT_N` ( or some other string that can be used to interpret what is being done here )

Reply:
sure, I'll change it to ... maybe `DANGEROUS_PERCENT_N` sounds appropriate just to drive the point home? :laughing:

Reply:
WFM